ksql-engine/src/main/java/io/confluent/ksql/security/KsqlSecurityContext.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2019 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.security;
+
+import io.confluent.ksql.services.ServiceContext;
+
+import java.security.Principal;
+import java.util.Optional;
+
+/**
+ * A class that provides KSQL security related information for KSQL user requests.
+ */
+public class KsqlSecurityContext {
+  private final Optional<Principal> userPrincipal;
+  private final ServiceContext serviceContext;
+
+  public KsqlSecurityContext(Optional<Principal> userPrincipal, ServiceContext serviceContext) {
+    this.userPrincipal = userPrincipal;
+    this.serviceContext = serviceContext;
+  }
+
+  /**
+   * Returns a {@code java.security.Principal} object containing the name of the current
+   * authenticated user. If the user has not been authenticated, the method returns
+   * {@code Optional.empty}.
+   *
+   * @return a {@code java.security.Principal} containing the name of the user making this request;
+   * {@code Optional.empty} if the user has not been authenticated
+   */
+  public Optional<Principal> getUserPrincipal() {
```

Comment:
reading the PR again.. just want to confirm that the main purpose of introducing `KsqlSecurityContext` is to have the user name part of the caching key?

Reply:
Yes.

I initially wanted to wrap the user name into the `ServiceContext`. But that class has a constructor with too many parameters, and I didn't want to add a new one. Also, seems having a `KsqlSecurityContext` might be beneficial in the future if we want to add extra security configurations there, such as ssl certificates, session tokens, etc.